# Sensor Fuser Loader
estimation:
  # sensor plugin
  sensor_list:
  - sensor_plugin/imu
  - sensor_plugin/mocap
  - sensor_plugin/alt
  - sensor_plugin/vo
  - sensor_plugin/gps
  - sensor_plugin/plane_detection

  # egomotion estimation
  egomotion_list:
  - kalman_filter/kf_pos_vel_acc
  - kalman_filter/kf_pos_vel_acc
  - kalman_filter/kf_pos_vel_acc

  # experiment estimation
  experiment_list:
  - kalman_filter/kf_pos_vel_acc
  - kalman_filter/kf_pos_vel_acc
  - kalman_filter/kf_pos_vel_acc

  # specific id(int) of sensor fuser, axis for kf
  fuser_egomotion_id1: 8 # 1 << 3
  fuser_egomotion_id2: 16 # 1 << 4
  fuser_egomotion_id3: 32 # 1 << 5

  fuser_experiment_id1: 8 # 1 << 3
  fuser_experiment_id2: 16 # 1 << 4
  fuser_experiment_id3: 32 # 1 << 5

  # specific name of sensor fuser
  fuser_egomotion_name1: /ee/x
  fuser_egomotion_name2: /ee/y
  fuser_egomotion_name3: /ee/z

  fuser_experiment_name1: /ex/x
  fuser_experiment_name2: /ex/y
  fuser_experiment_name3: /ex/z


# EGOMOTION_ESTIMATION_MODE = 0
# EXPERIMENT_MODE = 1
# GROUND_TRUTH_MODE = 2

sensor_plugin:
  imu:
    estimate_mode: 3 # 1<<0 + 1<<1
    level_acc_bias_noise_sigma: 0.0001
  mocap:
   estimate_mode: 6 # 1<<1 + 1<<2
  alt:
    estimate_mode: 1 # 1<<0
  gps:
    estimate_mode: 1 # 1<<0
  vo:
    estimate_mode: 1 # 1<<0
    sensor_frame: realsense1_pose_frame
    downwards_vo_min_height: 0.4
  vo1: #downward
    vo_sub_topic_name: realsense1/odom/throttle
  plane_detection:
    estimate_mode: 1 # 1<<0
    sensor_frame: rs_d435_color_optical_frame
    variable_sensor_tf_flag: true
    plane_detection_sub_topic_name: multi_plane_sac_segmentation/output_coefficients

